src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -375,6 +374,39 @@ def process_inputs(self, inputs: TextGenerationInput) -> List[numpy.ndarray]:
         :param inputs: the input schema for the pipeline
         :return: the inputs for the engine
         """
+        if not self.cache_support_enabled and inputs.max_tokens > 1:
+            raise ValueError(
+                "The model used for inference does not support kv cache. It is "
+                "assumed that it maps from the token sequence to predicted logits."
+                "Set `max_tokens` to 1 to support that scenario."
+            )
+
+        # If the num_generated_predictions > 1, check how many times each prompt given
+        # as an input was repeated. If just once, the number of generated predictions
+        # for that particular prompt is equal to num_generated_predictions. This is done
+        # by repeating the prompt num_generated_predictions times If that
+        # prompt was given as an input multiple times, ignore the value of
+        # num_generated_predictions and produce n predictions, where n is the number of
```

Comment:
is there a reason we don't postprocess the generated sequences and group them by the original sequence idx now with multiple generated values? that way the user has a 1:1 mapping between their actual inputs and what they get back

Reply:
Do you mean in general when the user provides multiple input sequences which are identical?